### ISSUE TITLE
"IOError: cannot write mode P as JPEG" when creating an asset via /api/0.1/assets/

### ISSUE THREAD

**Issue by ghing** [denverfoundation/storybase#726]
```
Traceback (most recent call last):

  File "/home/floodlight/webapps/floodlight_production/venv/lib/python2.7/site-packages/tastypie/resources.py", line 202, in wrapper
    response = callback(request, *args, **kwargs)

  File "/home/floodlight/webapps/floodlight_production/venv/lib/python2.7/site-packages/tastypie/resources.py", line 441, in dispatch_list
    return self.dispatch('list', request, **kwargs)

  File "/home/floodlight/webapps/floodlight_production/floodlight/apps/storybase/api/resources.py", line 428, in dispatch
    response = method(request, **kwargs)

  File "/home/floodlight/webapps/floodlight_production/venv/lib/python2.7/site-packages/tastypie/resources.py", line 1264, in post_list
    updated_bundle = self.full_dehydrate(updated_bundle)

  File "/home/floodlight/webapps/floodlight_production/venv/lib/python2.7/site-packages/tastypie/resources.py", line 745, in full_dehydrate
    bundle.data[field_name] = method(bundle)

  File "/home/floodlight/webapps/floodlight_production/floodlight/apps/storybase_asset/api.py", line 253, in dehydrate_thumbnail_url
    return bundle.obj.get_thumbnail_url(width=222, height=222)

  File "/home/floodlight/webapps/floodlight_production/floodlight/apps/storybase_asset/models.py", line 553, in get_thumbnail_url
    thumbnail = thumbnailer.get_thumbnail(thumbnail_options)

  File "/home/floodlight/webapps/floodlight_production/venv/lib/python2.7/site-packages/easy_thumbnails/files.py", line 396, in get_thumbnail
    thumbnail = self.generate_thumbnail(thumbnail_options)

  File "/home/floodlight/webapps/floodlight_production/venv/lib/python2.7/site-packages/easy_thumbnails/files.py", line 302, in generate_thumbnail
    quality=quality).read()

  File "/home/floodlight/webapps/floodlight_production/venv/lib/python2.7/site-packages/easy_thumbnails/engine.py", line 55, in save_image
    image.save(destination, format=format, **options)

  File "/home/floodlight/webapps/floodlight_production/venv/lib/python2.7/site-packages/PIL/Image.py", line 1439, in save
    save_handler(self, fp, filename)

  File "/home/floodlight/webapps/floodlight_production/venv/lib/python2.7/site-packages/PIL/JpegImagePlugin.py", line 425, in _save
    raise IOError("cannot write mode %s as JPEG" % im.mode)

IOError: cannot write mode P as JPEG

```

I don't think this affects many images.  SmileyChris/easy-thumbnails#95 gives a pretty good description of what's going on here.  I'm going to try to get the image in question to troubleshoot further.

**Comment by ghing** [comment]
Just as a sanity check, I created a PNG version of the `test_image.jpg` file included in the repo and it uploaded without error.  It seems that this issue is indeed tied to something specific about this image.

**Comment by ghing** [comment]
I also tried re-saving the image using GIMP and found that, just opening the file and saving it using the program's export feature, produced an image with the same issue.

I then opened the image and changed its color mode from indexed to RGB and saved a copy of the file.  The RGB version uploaded fine.

We should figure out what is going on with indexed PNGs, but the workaround is to just convert it to use RGB colors instead of indexed colors.
